project/Build.scala

```diff
@@ -810,48 +810,53 @@ object Build {
   ).zippedSettings("library")(
       commonLinkerSettings _
   ).settings(
-      sourceGenerators in Compile += Def.task {
-        val dir = (sourceManaged in Compile).value
-        val privateLibProducts = (products in (linkerPrivateLibrary, Compile)).value
+      if (isGeneratingForIDE) {
+        unmanagedSourceDirectories in Compile +=
+          baseDirectory.value.getParentFile.getParentFile / "js/src/main/scala-ide-stubs"
+      } else {
+        sourceGenerators in Compile += Def.task {
```

Comment:
Have you considered introducing a helper `sourceGeneratorOrStub` or similar?

This might make it easier for us to spot sites where we use source generators without IDE support.

Reply:
I'm not sure the abstraction is worth it TBH. Spotting sites with source generators without IDE support is easy enough: grep for `sourceGenerators`, and if it's not enclosed in an `if (generatingForIDE)` test, it doesn't have an IDE stub.

There is actually another one like that for the `SourceMapTest`. But since that one isn't used by other code (it's just a test), it doesn't need any IDE stub.